File: 1.TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  for (let i = 0; i < nums.length; i++) {
    let toFind = nums.indexOf(target - nums[i]);
    if (toFind != -1 && toFind != i) {
      return [i, toFind]
    }
  }
  return null;
};

File: 1.TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
       map<int,int> input_map;
       for(int i = 0; i < nums.size(); i++){
         if(input_map.find(target - nums[i]) != input_map.end())
           return {input_map[target-nums[i]],i};
         else
           input_map[nums[i]] = i;
       }
       return {-1, -1};
    }
};
